libkbfs/bare_root_metadata_v2.go

```diff
@@ -379,24 +359,39 @@ func (md *BareRootMetadataV2) makeSuccessorCopyV3(ctx context.Context, config Co
 	*BareRootMetadataV3, ExtraMetadata, error) {
 	mdCopy := &BareRootMetadataV3{}
 
-	// Fill out the writer metadata and new writer key bundle.
-	wkb, err := md.WriterMetadataV2.ToWriterMetadataV3(
-		ctx, config.Codec(), config.Crypto(), config.KeyManager(), kmd, &mdCopy.WriterMetadata)
-	if err != nil {
-		return nil, nil, err
-	}
+	// Fill out the writer metadata.
+	mdCopy.WriterMetadata = md.WriterMetadataV2.ToWriterMetadataV3()
 
-	var rkb *TLFReaderKeyBundleV3
+	// Have this as ExtraMetadata so we return an untyped nil
+	// instead of a typed nil.
+	var extraCopy ExtraMetadata
 	if md.LatestKeyGeneration() != PublicKeyGen {
-		// Fill out the reader key bundle.
-		rkb, err = md.RKeys.ToTLFReaderKeyBundleV3(config.Codec(), wkb)
+		// Fill out the writer key bundle.
+		wkb, err := md.WKeys.ToTLFWriterKeyBundleV3(
```

Comment:
This makes more sense to me to put this here, as opposed to in ToWriterMetadataV3.